docs/security/threat-model.md

```diff
@@ -1,223 +1,303 @@
+
+### Threat 11 - Attacker deploys workloads to namespaces that are exempt from admission control
+
+#### Scenario
 
-## Threat 11 - Attacker deploys workloads to namespaces that are  exempt from admission control
 An attacker is able to deploy workloads to Kubernetes namespaces that are exempt
 from the admission controller configuration.
 
-**Mitigation**
-RBAC rights are strictly controlled
+#### Mitigation: RBAC rights are strictly controlled
+
+#### To-do
 
-**To do**
 Most of the RBAC is out of scope with respect to this decision. However, the Kubewarden team aims to:
+
 - Warn users via our docs and *suggest* some minimum RBAC to be used.
 - Provide a policy which detects RBAC changes and **maybe** block them.
 
-## Threat 12 - Block rule can be bypassed due to missing match (e.g.  missing initcontainers)
+### Threat 12 - Block rule can be bypassed due to missing match (for example, missing initcontainers)
+
+#### Scenario
+
 An attacker created a workload manifest which uses a feature of the Kubernetes
 API which is not covered by the admission controller
 
-**Mitigation**
+#### Mitigation
+
 All rules are reviewed and tested.
 
-**To do**
+#### To-do
+
 Introduce tests to cover this rule.
-As always, carefully review PRs changing the rules in the policies deployment.
+As always, you should review PRs changing the rules in the policies deployment.
+
+### Threat 13 - Attacker exploits bad string matching on a blocklist to bypass rules
+
+#### Scenario
 
-## Threat 13 - Attacker exploits bad string matching on a blocklist to  bypass rules
 An attacker, who has rights to create workloads, bypasses a rule by exploiting
 bad string matching.
 
-**Mitigation**
-All rules are reviewed and tested.
+#### Mitigation
+
+All rules should be reviewed and tested.
+
+#### To-do
 
-**To do**
 Introduce tests to cover this rule.
-As always, carefully review PRs changing the rules in the policies deployment.
+As always, you should review PRs changing the rules in the policies deployment.
+
+### Threat 14 - Attacker uses new/old features of the Kubernetes API which have no rules
+
+#### Scenario
 
-## Threat 14 - Attacker uses new/old features of the Kubernetes API  which have no rules
 An attacker, with rights to create workloads, uses new features of the Kubernetes
 API (for example, a changed API version) to bypass a rule.
 
-**Mitigation**
+#### Mitigation
+
 All rules are reviewed and tested.
 
-**To do**
+#### To-do
+
 Introduce tests to cover this rule.
-Create a configuration to reject requests where the API version not
-cover by the policy by default. Kubewarden should warn policy developers to cover all the
-supported API version in their tests and reject all others.
+Create a configuration to reject requests where the API version doesn't cover by the policy by default.
+Kubewarden should warn policy developers to cover all the supported API version in their tests and reject all others.
```

Comment:
We now have a policy that flags deprecated resources being used: https://github.com/kubewarden/deprecated-api-versions-policy

We could drop this TODO and write something like that under the mitigation:

> Rely on [deprecated-api-versions-policy](https://github.com/kubewarden/deprecated-api-versions-policy) to identify and block usage of deprecated resources.

Reply:
Done, Thanks.

Reply:
Unresolving.

@flavio Note that the deprecated-api-versions-policy only deals with CRs known to it.
The threat is both deprecated api versions, and new unknown versions that are misused, hence the policy only covers half of it.

For it to be covered fully, we should add an allowlist to deprecated-api-versions-policy, or create a new allowlist-api-versions-policy. This is covered in https://github.com/kubewarden/kubewarden-controller/issues/227.